library/psa_crypto.c

```diff
@@ -7076,6 +7076,84 @@ psa_status_t psa_key_derivation_input_key(
 }
 
 
+psa_status_t psa_key_derivation_verify_bytes(
+    psa_key_derivation_operation_t *operation,
+    const uint8_t *expected_output,
+    size_t output_length)
+{
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+    uint8_t actual_output[output_length];
+
+    status = psa_key_derivation_output_bytes(operation,
+                                             actual_output,
+                                             output_length);
+    if (status != PSA_SUCCESS) {
+        goto exit;
+    }
+
+    if (mbedtls_ct_memcmp(expected_output, actual_output,output_length) != 0) {
+        status = PSA_ERROR_INVALID_SIGNATURE;
+        goto exit;
+    }
+
+exit:
+    mbedtls_platform_zeroize(actual_output, output_length);
+    if (status != PSA_SUCCESS && status != PSA_ERROR_INVALID_SIGNATURE &&
```

Comment:
The documentation says [call `psa_key_derivation_abort` when the errors are not insufficient data or invalid signature](https://arm-software.github.io/psa-api/crypto/1.1/api/ops/kdf.html#c.psa_key_derivation_verify_key:~:text=cannot%20be%20exported.-,If%20this%20function%20returns%20an%20error%20status%20other%20than%20PSA_ERROR_INSUFFICIENT_DATA%20or%20PSA_ERROR_INVALID_SIGNATURE%2C%20the%20operation%20enters%20an%20error%20state%20and%20must%20be%20aborted%20by%20calling%20psa_key_derivation_abort().,-Note). Should `psa_key_derivation_abort` be called whenever `status != PSA_SUCCESS`?

Reply:
Good question! [I'm not sure](https://github.com/ARM-software/psa-api/issues/112).